src/Wrapper/Algorithm.cs

```diff
@@ -47,6 +47,91 @@ public static Complex SumAll(Array arr)
             return new Complex(r, i);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static object MinAll<T>(Array arr)
+        {
+            af_dtype dtype = Internal.toDType<T>();
+            double r, i;
+            switch(dtype)
+            {
+                case af_dtype.f32:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (float)r;
+                case af_dtype.f64:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (double)r;
+                case af_dtype.s32:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (int)r;
+                case af_dtype.s64:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (long)r;
+                case af_dtype.u32:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (uint)r;
+                case af_dtype.u64:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (ulong)r;
+                case af_dtype.u8:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (byte)r;
+                case af_dtype.s16:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (short)r;
+                case af_dtype.u16:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out _, arr._ptr));
+                    return (ushort)r;
+                case af_dtype.c32:
+                case af_dtype.c64:
+                    Internal.VERIFY(AFAlgorithm.af_min_all(out r, out i, arr._ptr));
+                    return new Complex(r, i);
+                default:
+                    throw new NotSupportedException("Data type not supported");
+            }
```

Comment:
This logic can be squashed down further.
```c#
af_dtype dtype = Internal.toDType<T>();
if ( <dtype is none of the required enums> ) {
  throw new NotSupportedException("Data type not supported");
}
double r, i;
Internal.VERIFY(AFAlgorithm.af_min_all(out r, out i, arr._ptr));
if (dtype == c32 || dtype == c64) {
    return new Complex(r, i);
} else {
    return (T)r;
}
```

Reply:
That's great, thanks for catching this.